managed/src/main/java/com/yugabyte/yw/controllers/CertificateController.java

```diff
@@ -128,13 +129,34 @@ public Result getRootCert(UUID customerUUID, UUID rootCA) {
     }
   }
 
-  public Result list(UUID customerUUID) {
-    List<CertificateInfo> certs = CertificateInfo.getAll(customerUUID);
-    if (certs == null) {
-      return ApiResponse.error(BAD_REQUEST, "Invalid Customer UUID: " + customerUUID);
-    }
-    return ApiResponse.success(certs);
-  }
+	public Result list(UUID customerUUID) {
+		List<CertificateInfo> certs = CertificateInfo.getAll(customerUUID);
+		HashMap<String, Boolean> removable = new HashMap<String, Boolean>();
+		List<Universe> universes = Universe.getAll();
+		for (Universe universe : universes) {
+			try {
+				Universe universe_obj = Universe.get(universe.universeUUID);
+				UUID cert = universe_obj.getUniverseDetails().rootCA;
+				removable.put(cert.toString(), true);
+			} catch (NullPointerException a) {
+				continue;
+			}
+		}
+		JsonNode certificates = Json.toJson(certs);
+		for (JsonNode cert : certificates) {
+			JsonNode cert_uuid = cert.get("uuid");
+			Boolean value = removable.get(cert_uuid.asText());
```

Comment:
You never put false into the map in the above loop.
So when `value` exists it can only be `true`. 
Yet here you are saying if value exists always set removable to `false`.

I think there is a bug if meaning of  removable remains same between the two places.

Or did you mean to call the map inUse? Please clarify.

Reply:
Okay check the description. The map above is actually saying removable true when certificate is attached to universe.
`HashSet<String> inUseCerts ...
<first loop to populate this set>
`
Then In second loop you just do:
`certs.put("inUse", inUseCerts.contains(cert.get("uuid").asText()))
`

Reply:
Moved it to hashset.

Reply:
Dont see this. I still see it is HashMap above and still named removable.

Reply:
The PR is still pointing to the old branch. Opening we are giving PR from Jaydeep's repo.